src/sentry/queue/routers.py

```diff
@@ -0,0 +1,103 @@
+import random
+from collections.abc import Iterator, Mapping, Sequence
+from itertools import cycle
+from typing import Any, NamedTuple
+
+from django.conf import settings
+
+from sentry import options
+from sentry.celery import app
+
+
+def _get_known_queues() -> set[str]:
+    return {c_queue.name for c_queue in app.conf.CELERY_QUEUES}
+
+
+def _validate_destiantions(destinations: Sequence[str]) -> None:
+    for dest in destinations:
+        assert dest in _get_known_queues(), f"Queue {dest} in split queue config is not declared."
+
+
+class SplitQueueRouter:
+    """
+    Returns the split queue to use for a Celery queue.
+    Split queues allow us to spread the load of a queue to multiple ones.
+
+    This takes in input a queue name and returns the split. It is supposed
+    to be used by the code that schedules the task.
+
+    Each split queue can be individually rolled out via options.
+
+    WARNING: Do not forget to configure your workers to listen to the
+    queues appropriately before you start routing messages.
+    """
+
+    def __init__(self) -> None:
+        known_queues = _get_known_queues()
+        self.__queue_routers = {}
+        for source, destinations in settings.CELERY_SPLIT_QUEUE_ROUTES.items():
+            assert source in known_queues, f"Queue {source} in split queue config is not declared."
+            _validate_destiantions(destinations)
```

Comment:
```suggestion
            _validate_destinations(destinations)
```
To go with Joris' suggestion.

Reply:
I thought I had installed the spell checker in VSCode.